### ISSUE TITLE
URI.escape is obsolete in Ruby 2.7

### ISSUE THREAD

**Issue by lenon** [oracle/oci-ruby-sdk#56]
Hi,
Ruby 2.7 made `URI.escape` obsolete. When running the latest version of the SDK with Ruby 2.7, the following warning appears:

```
~/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/oci-2.10.0/lib/oci/api_client.rb:582: warning: URI.escape is obsolete
```

It seems to be caused by the following line:

https://github.com/oracle/oci-ruby-sdk/blob/416f4594130cc6c5e53eedc94bc795add7bc6629/lib/oci/api_client.rb#L582

`URI.encode` is an alias to `URI.escape`. It may be possible to replace it by `URI.encode_www_form_component`, but it seems to be necessary to split the query string part and encode it only.

The following blog post explains the issue really well: https://docs.knapsackpro.com/2020/uri-escape-is-obsolete-percent-encoding-your-query-string

**Comment by ZiyaoQiao** [comment]
Hi @lenon thanks for letting us know, we'll fix this issue in our next release. Thanks

**Comment by ZiyaoQiao** [comment]
@lenon We've released Ruby 2.11.0 yesterday with this fix. I'll close this issue, feel free to reopen if you're still facing any issue. Thanks
